packages/@glimmer/compiler/lib/template-compiler.ts

```diff
@@ -81,7 +81,17 @@ export default class TemplateCompiler {
       }
     }
 
-    if (hasSplat) {
+    if (isDynamicComponent(action)) {
+      let head: string | 0, rest: string[];
```

Comment:
Is there a better type we can use here than `string | 0`? This took me a few attempts to read to understand what was going on, particularly since the zero literal type looked like a bitwise operation at first.

Reply:
Ah, yes, great point.  This was a bit of fumbling on my part actually. We actually expect either `0` (which is what `this` is represented as) or the string for the "head" of the path.  I'll try to do some tweaking to make this type more intention revealing...